### PR TITLE
`/eth/v1/beacon/blinded_blocks/{block_id}` API endpoint

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements fetching blinded blocks through the Beacon API as described in https://github.com/ethereum/beacon-APIs/pull/241. See also https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlindedBlock.

**Other notes for review**

This PR also adds missing `ExecutionOptimistic` field to the response of non-blinded SSZ version and improves a few small things in existing grpc endpoints.

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 1]
```diff
@@ -634,8 +671,277 @@ func (bs *Server) GetBlockSSZV2(ctx context.Context, req *ethpbv2.BlockRequestV2
...
+	}
+	// ErrUnsupportedGetter means that we have another block type
+	if !errors.Is(err, blocks.ErrUnsupportedGetter) {
+		return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
+	}
+
+	bellatrixBlk, err := blk.PbBellatrixBlock()
+	if err == nil {
+		if bellatrixBlk == nil {
+			return nil, status.Errorf(codes.Internal, "Nil block")
```
Comment (terencechain) [inline beacon-chain/rpc/eth/beacon/blocks.go:878]:
```suggestion
			return nil, status.Error(codes.Internal, "Nil block")
```

Comment (terencechain) [reply beacon-chain/rpc/eth/beacon/blocks.go:878]: Same in other instances

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 2]
```diff
@@ -634,8 +671,277 @@ func (bs *Server) GetBlockSSZV2(ctx context.Context, req *ethpbv2.BlockRequestV2
...
+			return nil, status.Errorf(codes.Internal, "Could not convert block to blinded block: %v", err)
+		}
+		blindedBellatrixBlock, err := blindedBlkInterface.PbBlindedBellatrixBlock()
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
+		}
+		v2Blk, err := migration.V1Alpha1BeaconBlockBlindedBellatrixToV2Blinded(blindedBellatrixBlock.Block)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
+		}
```
Comment (terencechain) [inline beacon-chain/rpc/eth/beacon/blocks.go:891]: why convert to blind?

Comment (rauljordan) [reply beacon-chain/rpc/eth/beacon/blocks.go:891]: This endpoint is requesting a blinded block, so we must convert a bellatrix block to blinded format

**Review by rauljordan — COMMENTED** [review]
I reviewed @rkapka and aside from the minor suggestions from Terence, it looks good to merge

File: beacon-chain/rpc/eth/beacon/blocks.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/rpc/eth/beacon/blocks.go:891]: Ah. Thanks

**Review by terencechain — DISMISSED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**